drivers/d3d12/rendering_device_driver_d3d12.cpp

```diff
@@ -3918,6 +4028,22 @@ void RenderingDeviceDriverD3D12::command_uniform_set_prepare_for_use(CommandBuff
 					for (uint32_t i = 0; i < tex_info->layers; i++) {
 						for (uint32_t j = 0; j < tex_info->mipmaps; j++) {
 							uint32_t subresource = D3D12CalcSubresource(tex_info->base_mip + j, tex_info->base_layer + i, 0, tex_info->desc.MipLevels, tex_info->desc.ArraySize());
+
+							// If about to sample a texture whose contents are still garbage, black it out.
+							// This is not the most optimal way to handle this, but should happen once in a blue moon.
+							if ((wanted_state & D3D12_RESOURCE_STATE_ALL_SHADER_RESOURCE) && tex_info->states_ptr->subresource_states[subresource].init_state != ResourceInfo::States::ResourceStateEx::INIT_STATE_DONE) {
+								TextureSubresourceRange range;
+								range.base_mipmap = j;
+								range.mipmap_count = 1;
+								range.base_layer = i;
+								range.layer_count = 1;
+								_command_clear_color_texture(p_cmd_buffer, tex_info, Color(), range, false);
```

Comment:
Not necessarily a code error just yet but it seems the comment is a bit misleading as to how often this happens. Upon running a very simple project I got this path above 3000 times on startup. So much that it triggered the error where it ran out of RTV descriptors and I had no choice but to increase it.

Probably worth reviewing how often this triggers and if it's intentional.

Reply:
The new approach should be free from that issue.